x-pack/plugins/apm/server/lib/helpers/get_document_sources.ts

```diff
@@ -231,3 +184,51 @@ export async function getDocumentSources({
     hasDurationSummaryField: false,
   });
 }
+
+async function getDurationSummarySupportByDocType({
+  apmEventClient,
+  documentTypes,
+  filters,
+  enableContinuousRollups,
+}: {
+  apmEventClient: APMEventClient;
+  documentTypes: ApmDocumentType[];
+  filters: estypes.QueryDslQueryContainer[];
+  enableContinuousRollups: boolean;
+}) {
+  const sources = documentTypes.flatMap((documentType) => {
+    const docTypeConfig = getConfigForDocumentType(documentType);
+
+    return (
+      enableContinuousRollups
+        ? docTypeConfig.rollupIntervals
+        : [RollupInterval.OneMinute]
+    ).flatMap((rollupInterval) => ({
+      documentType,
+      rollupInterval,
+      query: getRequest({
+        documentType,
+        rollupInterval,
+        filters: [...filters, getDurationLegacyFilter()],
+      }),
+    }));
+  });
+
+  const allResponses = (
+    await apmEventClient.msearch(
+      'get_duration_summary_support',
+      ...sources.map((source) => source.query)
+    )
+  ).responses;
+
+  return sources.reduce(
+    (acc, curr, index) => ({
+      ...acc,
+      [curr.documentType]: {
+        ...acc[curr.documentType],
+        [curr.rollupInterval]: allResponses[index].hits.total.value === 0,
+      },
+    }),
+    {} as { [key: string]: { [key: string]: boolean } }
+  );
```

Comment:
**suggestion:** on the effort to reduce pressure on GC done with this work, we could reduce these nested spreading to something like:
```ts
return sources.reduce(
  (acc, curr, index) => {
    const {documentType, rollupInterval} = curr
    acc[documentType] = { 
      ...acc[documentType], 
      [rollupInterval]: allResponses[index].hits.total.value === 0 
    }

    return acc;
  },
  {} as Record<string, Record<string, boolean>
);
```

Reply:
I think it's a good idea to do this but only for consistency. this is like 10 iterations.

Reply:
Nice catch! Thanks.